openpgp/packet/public_key.go

```diff
@@ -418,6 +418,9 @@ func (pk *PublicKey) parse(r io.Reader) (err error) {
 			return err
 		}
 		err = pk.edk.check()
+		if err == nil {
+			pk.PublicKey = ed25519.PublicKey(pk.edk.p.bytes[1:])
+		}
```

Comment:
does it matter that `pk.PublicKey` is not set in any of the cases above? (ElGamal,DSA,RSA*)

can you remind me how it is used by API consumers or internally?

Reply:
`pk.PublicKey` is set by `parseRSA`, `parseDSA` and `parseElGamal`

Reply:
Is is used by API Consumers, aka myself. PublicKey is a public field for PublicKey struct: https://github.com/keybase/go-crypto/blob/master/openpgp/packet/public_key.go#L238 and PubKeyAlgoEdDSA is the only alg which doesn't set the `PublicKey` field

Reply:
right, I'm curious what was breaking because of this field being nil after parsing the key.

Reply:
OK I'm starting to see what's up here. Our EdDSA implementation code here was not that great, and it's probably written by me, too. Thanks for working on this

Reply:
> right, I'm curious what was breaking because of this field being nil after parsing the key.

We use it at $WORK for interoperability between PGP and JOSE. The OpenPGP key parser exposes raw keys which are then used by the JOSE library, except for EdDSA.